src/api.rs

```diff
@@ -1710,12 +1745,23 @@ impl<T: Pixel> ContextInner<T> {
       .map(|(&output_frameno, _)| output_frameno)
       .ok_or(EncoderStatus::NeedMoreData)?; // TODO: doesn't play well with the below check?
 
-    if !self.needs_more_frames(
-      self.frame_invariants[&self.output_frameno].input_frameno
-    ) {
+    let input_frameno = self.frame_invariants[&self.output_frameno].input_frameno;
+    if !self.needs_more_frames(input_frameno) {
       return Err(EncoderStatus::LimitReached);
     }
 
+    // Block importance computation needs to have FI's already processed `LOOKAHEAD_FRAMES` frames into the future.
+    let ready_frames = self
```

Comment:
The iterator seems the same as the one used in line 1494, please factor it out.

Reply:
The borrow checker doesn't like that, because it's referencing `self.output_frameno` within the iterator. I could make it its own function if I make it collect into a `Vec`, but that would be less performant, because in one of the usages we only `count` the items in the iterator without needing to collect.